packages/build/src/plugins_core/deploy/grpc_deploy.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const { promisify } = require('util')
+
+const {
+  loadPackageDefinition,
+  credentials: { createInsecure: createCredentials },
+} = require('@grpc/grpc-js')
+const { load } = require('@grpc/proto-loader')
+
+const { addErrorInfo } = require('../../error/info')
+const { logDeploySuccess } = require('../../log/messages/plugins')
+const { runsAfterDeploy } = require('../../plugins/events')
+
+const SERVICE_PROTO = `${__dirname}/build_service.proto`
+
+// Send a gRPC request to the buildbot to deploy the site
+const grpcDeploy = async function ({ buildbotServerSocket, events, logs }) {
+  const client = await startClient(buildbotServerSocket)
+  try {
+    await waitForReady(client)
+    await runPostDeploy(client, events)
+    logDeploySuccess(logs)
+  } finally {
+    client.close()
+  }
+}
+
+// Start the gRPC client
+const startClient = async function (buildbotServerSocket) {
+  const serviceDefinition = await load(SERVICE_PROTO)
```

Comment:
Load `*.proto` at runtime since it takes less than 10ms. As a performance improvement, we could use `protoc` to parse the `*.proto` as build-time, but this introduces a build step which `@netlify/build` does not have yet, for a performance improvement that is very low.